src/zulqer/finalDirection/finalDirection.ts

```diff
@@ -0,0 +1,101 @@
+enum validDirections {
+  NORTH,
+  EAST,
+  SOUTH,
+  WEST
+}
+enum validMoves {
+  STRAIGHT,
+  RIGHT,
+  LEFT,
+  BACK
+}
+const MOVE_REWARD = {
+  STRAIGHT: 0,
+  RIGHT: 1,
+  LEFT: -1,
+  BACK: 2
+};
+const numberOfDirection = 4;
+
+const SAMPLE_VALID_INPUT: Array<validMoves> = [
+  validMoves.STRAIGHT,
+  validMoves.RIGHT,
+  validMoves.LEFT,
+  validMoves.BACK,
+  validMoves.RIGHT,
+  validMoves.STRAIGHT,
+  validMoves.LEFT
+];
+const SAMPLE_UNKNOWN_INPUT = [
+  'STRAIGHT',
+  'RIGHT',
+  'LEFT',
+  'BACK',
+  'RIGHT',
+  'STRAIGHT',
+  'LEFT'
+];
+const SAMPLE_INVALID_UNKNOWN_INPUT = [
+  'STRAIGHT',
+  'RIGHT',
+  'LEFT',
+  'BACK',
+  'RIGHT',
+  'STRAIGHT',
+  'LEFT',
+  'UP'
+];
+const isValidMove = (move: string): boolean => {
+  return Object.keys(validMoves).includes(move);
+};
+const isValidDirection = (direction: string): boolean => {
+  return Object.keys(validDirections).includes(direction);
+};
+
+class FinalDirection {
+  getFinalDirectionByValidatedInput = (
+    moves: Array<validMoves>,
+    initialFacing: validDirections = validDirections.NORTH
+  ): string => {
+    const sumOfRewards = moves
+      .map((move) => MOVE_REWARD[validMoves[move]])
+      .reduce((sum, currentValue) => sum + currentValue, initialFacing);
+
+    return validDirections[sumOfRewards % numberOfDirection];
+  };
+
+  getFinalDirectionByUnknownInput = (
+    moves: string[] = [],
+    initialFacing = 'NORTH'
+  ): string => {
```

Comment:
Why two different implementations? Please figure out the best one and try to remove the other.